sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -770,11 +838,15 @@ cdef class DistanceMetric{{name_suffix}}:
         Y_is_sparse = issparse(Y)
 
         if not X_is_sparse and not Y_is_sparse:
+            X = np.asarray(X, order='C')
+            Y = np.asarray(Y, order='C')
```

Comment:
I do not think we need this, because `_pairwise_dense_dense` already casts `X` and `Y` to the correct dtype and order.


```suggestion
```